fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/query/FDBSimpleQueryGraphTest.java

```diff
@@ -386,12 +386,12 @@ public void testPlanCrossProductJoin() throws Exception {
 
         final BindingMatcher<? extends RecordQueryPlan> planMatcher =
                 flatMapPlan(
-                        typeFilterPlan(scanPlan().where(scanComparisons(unbounded())))
-                                .where(recordTypes(containsAll(ImmutableSet.of("RestaurantReviewer")))),
                         descendantPlans(
                                 indexPlan()
                                         .where(indexName("RestaurantRecord$name"))
-                                        .and(scanComparisons(range("[[name],[name]]")))));
+                                        .and(scanComparisons(range("[[name],[name]]")))),
+                        typeFilterPlan(scanPlan().where(scanComparisons(unbounded())))
+                                .where(recordTypes(containsAll(ImmutableSet.of("RestaurantReviewer")))));
```

Comment:
Is there something more to this change than just reordering the arguments?

Reply:
This is directly related to the cost model tie-breaking on the plan hash.